### ISSUE TITLE
Incorrect handling of rpy triple near the rpy singularity 

### ISSUE THREAD

**Issue by traversaro** [ros/urdfdom_headers#18]
Commit https://github.com/ros/urdfdom_headers/commit/96b8a4ba5a06309232f03a018e70e929f89f93bc (in PR https://github.com/ros/urdfdom_headers/pull/11) introduced a regression in handling rpy triple with a pitch value of `M_PI/2` or `-M_PI/2`. 

In particular the rpy triple `0.0 -M_PI/2 0.0` is now converted (incorrectly) to the rotation matrix:
~~~ 
 0   0  -1
 0  -1   0
-1   0   0
~~~
instead of the correct: 
~~~
 0   0  -1
 0   1   0
 1   0   0
~~~

The regression is quite critical because it can cause software loading urdf with rpy triples like ``0.0 -M_PI/2 0.0` to load a corrupted model. The user risks to not understand that something is going wrong because no error message is generated. 

Downstream bug: https://github.com/robotology/idyntree/issues/79 .

**Comment by traversaro** [comment]
Correct me if I am wrong @j-rivero, but I guess that the bugged version (the one containing commit https://github.com/ros/urdfdom_headers/commit/96b8a4ba5a06309232f03a018e70e929f89f93bc ) was never shipped in the ubuntu repo, as the relative SRU ( https://bugs.launchpad.net/ubuntu/+source/urdfdom-headers/+bug/1471047 ) was never closed, right?

**Comment by j-rivero** [comment]
> Correct me if I am wrong @j-rivero, but I guess that the bugged version (the one containing commit 96b8a4b ) was never shipped in the ubuntu repo, as the relative SRU ( https://bugs.launchpad.net/ubuntu/+source/urdfdom-headers/+bug/1471047 ) was never closed, right?

Unfortunately this time, you are right.

**Comment by scpeters** [comment]
This was fixed by #19 and can be closed.
